### ISSUE TITLE
[BUG] - Wrong window focused with multiple clients

### ISSUE THREAD

**Issue by rpav** [Morph21/MercuryTrade-Community-Fork#123]
**Describe the bug**
When running multiple clients, e.g. one "trading character" and one "main character," the client focused for sending messages / invites / etc is incorrect.  

**To Reproduce**
1. Run 2 clients.
2. Use Mercury to do anything causing the client to be focused
3. A random client will be focused

**Expected behavior**
Either:
* The client corresponding to the message ought to be focused
* At a minimum, a single configurable client ought to be focused

**Desktop (please complete the following information):**
- Windows 10, DX11/DX12, latest release as of filing
  
**Additional context**

This appears to be a problem with `gameToFront()` in ChatHelper.java:204.  This iterates all the windows and finds one with the expected class ("POEWindowClass"), which ends up semi-randomly picking a window.

*However*, this should be pretty easy to fix... the proper PID for every message is in Client.txt!  One should be able to either check for the window's PID, or only iterate the proper client's windows, etc.  I personally would be happy enough with setting a single global pid at this point.

I'm filing this in case it's something that can be fixed relatively quickly.  I may try hacking out a solution this weekend, but I'm not really set up for java dev at the moment.   I'll file a PR if I get to it and have something not entirely a hack.

And thanks for maintaining this, it's incredibly useful.

**Comment by Morph21** [comment]
You probably could change this

```
if (SystemUtils.IS_OS_WINDOWS) {
            WindowUtils.getAllWindows(false).forEach(window -> {
                char[] className = new char[512];
                User32.INSTANCE.GetClassName(window.getHWND(), className, 512);
                if (Native.toString(className).equals("POEWindowClass")) {
                    User32.INSTANCE.ShowWindow(window.getHWND(), 5);

                    boolean isAtFront = User32.INSTANCE.SetForegroundWindow(window.getHWND());
                    int counter = 0;
                    while (!isAtFront && counter < 10) {
                        isAtFront = User32.INSTANCE.SetForegroundWindow(window.getHWND());
                        try {
                            Thread.sleep(10);
                        } catch (InterruptedException e) {
                            e.printStackTrace();
                        }
                        counter++;
                    }

                    User32.INSTANCE.SetFocus(window.getHWND());
                }
            });
```

into this:

```
if (SystemUtils.IS_OS_WINDOWS) {
            WindowUtils.getAllWindows(false).forEach(window -> {
                char[] className = new char[512];
                IntByReference i = new IntByReference();
                User32.INSTANCE.GetWindowThreadProcessId(window.getHWND(), i);
                if (i == 12345) {
                    User32.INSTANCE.ShowWindow(window.getHWND(), 5);

                    boolean isAtFront = User32.INSTANCE.SetForegroundWindow(window.getHWND());
                    int counter = 0;
                    while (!isAtFront && counter < 10) {
                        isAtFront = User32.INSTANCE.SetForegroundWindow(window.getHWND());
                        try {
                            Thread.sleep(10);
                        } catch (InterruptedException e) {
                            e.printStackTrace();
                        }
                        counter++;
                    }

                    User32.INSTANCE.SetFocus(window.getHWND());
                }
            });
```

replace 12345 with pid of your path of exile. It should work if you build it like that

**Comment by HiPoEGH** [comment]
Make a dropdown list?
Populate the list with all processes matching the "POEWindowClass", and let the user select one.

**Comment by LordBlick** [comment]
I can't give any code here because I'm rather Python developer, but I can put some kind of logic.
* At the first (because this is not solved yet) „Do not switch any window(old behaviour) ”should be option in UI – usually active window is properly selected manually plus  both characters frequently could to go to hideout, view betrayal cheat sheet etc.
* I think that low efford, temporary solution is just make an option with checkbox in UI „First running game instance is for trade one” to call window with lowest pid and login into trade character first, and another option – „Use it only with incoming message” – sometimes main account need to buy sth… ;)  

Of course detecting which window is active before any action also will be nice.
Current state (1.3.4) is more unplayable for me than before when manualy window switch was fine.

**Comment by Morph21** [comment]
Kinda fixed in 1.4.1 - you can disable bringing game to top in settings now, keep in mind that you need to focus game window where you want message to go before pressing it in MT ui. Probably later on will be added an option to select window which MT will bring to top but it's a lot of work so not for now
